lib/page/questions/widget/answer/answer_smiley_rating_bar_widget.dart

```diff
@@ -0,0 +1,74 @@
+import 'dart:math';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:survey/resource/dimens.dart';
+
+final selectedEmojiIndexProvider = StateProvider.autoDispose<int>((_) => 2);
+const _emojis = {
+  0: '😡',
+  1: '😕',
+  2: '😐',
+  3: '🙂',
+  4: '😄',
+};
+
+class AnswerSmileyRatingBarWidget extends ConsumerWidget {
+  final int itemCount;
+  final Function onRatingUpdate;
+
+  const AnswerSmileyRatingBarWidget({
+    required this.itemCount,
```

Comment:
I think we don't need to pass `itemCount` here, just use the `_emojis.length` it's because this is a smiley rating bar so we fixed the length here 🤔